core/store/src/db/rocksdb.rs

```diff
@@ -735,4 +738,27 @@ mod tests {
             }
         );
     }
+
+    #[test]
+    fn test_delete_range() {
+        let store = NodeStorage::test_opener().1.open().unwrap().get_store(crate::Temperature::Hot);
+        let keys = [vec![0], vec![1], vec![2], vec![3]];
+        let column = DBCol::Block;
+
+        let mut store_update = store.store_update();
+        for key in &keys {
+            store_update.insert(column, key, &vec![42]);
+        }
+        store_update.commit().unwrap();
+
+        let mut store_update = store.store_update();
+        store_update.delete_range(column, &keys[1], &keys[3]);
+        store_update.commit().unwrap();
+
+        let key_exists = |key_index: usize| store.get(column, &keys[key_index]).unwrap().is_some();
+        assert!(key_exists(0));
```

Comment:
Could we consider using `assert_matches` here, like `assert_matches!(store.get(column, &keys[1]), Ok(Some(_)))`? This would look more idiomatic, because it helps to avoid unwraps.

We can also use `assert_eq` and assert the exact values to check that they don't change.

Reply:
unwraps seems to be pretty common as part of tests in our codebase, could you please elaborate on why it should be avoided?

`assert_matches` is nightly-only, so unfortunately we can't use it: https://doc.rust-lang.org/std/assert_matches/macro.assert_matches.html

Reply:
Discussed offline + https://near.zulipchat.com/#narrow/stream/300659-Rust-.F0.9F.A6.80/topic/unwrap.20or.20match/near/320550691

+ note for further readers: we use a separate crate for assert_matches, not the one currently in std.

Reply:
@Longarithm thanks, I've updated the test to use 3rd party `assert_matches!`, please take a look

Reply:
@pugachAG Sorry for nitpicking again, but in your approach `assert_eq!` is enough now XD
Nice point that we actually have `exists` method though

Reply:
@Longarithm unfortunately `assert_eq!` results in compilation error here:
```
binary operation `==` cannot be applied to type `Result<bool, std::io::Error>`
```

Reply:
Oh, I didn't understand that before. This is... not what I expected from `io::Result`. Let's merge as is and I ask a question in Rust channel